src/audio/volume/volume_hifi3.c

```diff
@@ -88,8 +97,22 @@ static void vol_s24_to_s24_s32(struct comp_dev *dev, struct audio_stream *sink,
 
 			/* Store the output sample */
 			AE_S32_L_XC(out_sample, out, sizeof(ae_int32));
+#if CONFIG_IPC_MAJOR_4
```

Comment:
also to avoid the conditional macro here and other places it would be good to
```c
        /* Store the output sample */
        AE_S32_L_XC(out_sample, out, sizeof(ae_int32));

       /* calc peak vol */
       peak_vol_calc(context);
    }
}

/* update peak vol */
peak_vol_update(context);
```

where both ```peak_vol_update()``` and ```peak_vol_calc()``` are static inlines that are defined for CONFIG_PEAKVOL and defined empty for !CONFIG_PEAKVOL. This way the intrinsic loops are the same regardless of peak vol and gain format.

Reply:
updated, temporarily for IPC4 config, please check it